base/special/trig.jl

```diff
@@ -583,7 +583,7 @@ function atan(y::T, x::T) where T<:Union{Float32, Float64}
     #    S9) ATAN2(+-INF,-INF ) is +-3pi/4;
     #    S10) ATAN2(+-INF, (anything but,0,NaN, and INF)) is +-pi/2;
     if isnan(x) || isnan(y) # S1 or S2
```

Comment:
```suggestion
    if isnan(x) | isnan(y) # S1 or S2
```
`isnan(x) | isnan(y)` tends to compile nicer than `isnan(x) || isnan(y)`, at least on x86. For me, this saves 1 comparison and 1 branch.